content/what-is-jsx.md

```diff
@@ -0,0 +1,255 @@
+```
+
+or for a loop
+
+```jsx
+<List>
+    {items.map(item => (
+        <span>{item}</span>
+    ))}
+</List>
+```
+
+### Special Cases for Boolean Attributes
+
+Attributes without an explicit value in a JSX expression are interpreted as
+having the boolean value `true`. This means that the following two JSX blocks
+are identical:
+
+```jsx
+<button disabled>
+```
+
+```jsx
+<button disabled={true}>
+```
+
+### Fragment provided by JSX Implementation
+
+JSX always expects exactly one root element in an expression, but in some cases
+we want to return several at once. To do this, a JSX implementation will
+provide a `Fragment` component with provides an artificial hierarchy level.
+
+```jsx
+<dl>
+	{items.map(({ term, description }) => (
+		<Fragment>
+			<dt>{term}</dt>
+			<dd>{description}</dd>
+		</Fragment>
+	))}
+</dl>
+```
+
+### Trying it out
+
+You can try out how JSX translation works in the
+[Babel online compiler](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=DwWQngwg9gtgDlAdgU0QFwARoJZoDbIC8ARABLJ55QYDqUATngCbEB8AUAJDBytX3IYGbHADOAVxjAA9L3Yzw0eElRpWQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2&prettier=false&targets=&version=7.7.3&externalPlugins=).
+It translates it using the `createElement` function from the React library,
+but we can also tell Babel to use a different `createElement` function from a
+library like complate.
+
+
+The complate JSX implementation
+-------------------------------
+
+Now we want to look at the specifics of the complate JSX implementation.
+
+We saw before that
+
+```jsx
+<MyComponent title="Hello World">
+	<p>lorem ipsum</p>
+<MyComponent>
+```
+
+becomes
+
+```javascript
+createElement(MyComponent, { title: title },
+	createElement("p", null, description));
+```
+
+In order for this to work, the complate `createElement` function needs
+to be available in the JavaScript scope.
+
+`createElement` in complate expects a user defined component (called a macro)
+to be a function with the signature `(params, ...children)` and to return a JSX
+expression. An example definition for `MyComponent` could be:
+
+```jsx
+function MyComponent({ title }, ...children) {
+    return <article>
+        <h2>{title}</h2>
+        {children}
+    </article>;
+}
+```
+
+### Dealing with boolean attributes
+
+We saw above that JSX interprets an attribute without a value as being
+the same as the boolean value true. complate also ignores the blank
+values (`false`, `undefined`, `null`) so `<button disabled={undefined}>`
+and `<button disabled={null}>` generate the same `<button>` HTML Tag
+without the `disabled` property.
+
+complate will also ensure that boolean properties are correctly set on
+HTML elements. This means that `<button disabled={true}>` will be
+correctly translated to the HTML Tag `<button disabled>`.
```

Comment:
As mentioned elsewhere, this is a little tricky because it might be misunderstood: complate does _not_ know about boolean attributes the same way [it knows about void elements](https://github.com/complate/complate-stream/blob/77faf430a9cf04a9b19b1c623e6e72d7230e7f03/src/html.js#L6-L12), so it does not handle, say, [`aria-disabled`])(https://www.w3.org/TR/wai-aria-1.1/#aria-disabled) (which is _not- a boolean attribute). That's a subtle distinction lost on many, but this might be interpreted as complate attempting to remedy that situation, which it can't.